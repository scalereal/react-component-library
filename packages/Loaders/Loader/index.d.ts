import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
export declare const SizeMap: Props['size'][];
declare const Loader: React.FC<Props> & WithStyle;
export default Loader;
