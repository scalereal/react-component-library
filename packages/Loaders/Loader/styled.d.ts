import { Props } from '../types';
export declare const StyledLoader: import("styled-components").StyledComponent<"div", any, Props, never>;
export declare const LoaderWrapper: import("styled-components").StyledComponent<"div", any, Props, never>;
