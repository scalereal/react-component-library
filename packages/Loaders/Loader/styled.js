var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var StyledLoader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    vertical-align: middle;\n    pointer-events: none;\n    border-radius: 50%;\n    animation: 1s StyledLoader-01 linear infinite;\n\n    @keyframes StyledLoader-01 {\n        0% {\n            transform: rotate(0deg);\n        }\n        100% {\n            transform: rotate(360deg);\n        }\n    }\n    \n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    vertical-align: middle;\n    pointer-events: none;\n    border-radius: 50%;\n    animation: 1s StyledLoader-01 linear infinite;\n\n    @keyframes StyledLoader-01 {\n        0% {\n            transform: rotate(0deg);\n        }\n        100% {\n            transform: rotate(360deg);\n        }\n    }\n    \n"])));
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n            " + StyledLoader + "{\n                border: 0.188em solid transparent;\n            }";
        case "XS":
            return "font-size: 1rem;\n            " + StyledLoader + "{\n                border: 0.188em solid transparent;\n            }\n            ";
        case "S":
            return "font-size: 2rem;\n            " + StyledLoader + "{\n                border: 0.2em solid transparent;\n            }";
        case "M":
            return "font-size: 3rem;\n            " + StyledLoader + "{\n                border: 0.2em solid transparent;\n            }";
        case "L":
            return "font-size: 5rem;\n            " + StyledLoader + "{\n                border: 0.2em solid transparent;\n            }";
        case "XL":
            return "font-size: 6rem;\n            " + StyledLoader + "{\n                border: 0.2em solid transparent;\n            }";
        default:
            return "font-size: 1rem;\n        " + StyledLoader + "{\n            border: 0.15em solid transparent;\n        }";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n    ", "{\n        border-left-color:  ", ";\n        border-right-color:  ", ";\n    }\n"], ["\n    ", "\n    ", "\n    display: ", ";\n    ", "{\n        border-left-color:  ", ";\n        border-right-color:  ", ";\n    }\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.display ? 'inline-block' : 'none'; }, StyledLoader, function (pr) { return pr.Color ? pr.Color : 'inherit'; }, function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var templateObject_1, templateObject_2;
