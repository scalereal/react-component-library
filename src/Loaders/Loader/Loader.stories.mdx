import Loader from '.';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';
import { SizeMap } from '../types';

<Meta
    title="Compone       a {
            font-size: 1.5rem;
            text-decoration: none;

        }nts/Loaders/Loader"
    component={Loader}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader {...args} />
    </div>
);

<Preview>
    <Story name="Loader" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around' }}>
        <Loader size="S" Color="green" />
        <Loader size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <Loader size="XXS" />
        <Loader size="XS" />
        <Loader size="S" />
        <Loader size="M" />
        <Loader size="L" />
        <Loader size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader" />
