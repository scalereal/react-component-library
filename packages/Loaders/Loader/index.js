var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { LoaderWrapper, StyledLoader } from './styled';
export var SizeMap = ['XXS', 'XS', 'S', 'M', 'L', 'XL'];
var Loader = React.memo(React.forwardRef(function (props, ref) {
    var size = props.size;
    return (React.createElement(LoaderWrapper, __assign({ ref: ref, size: size }, props),
        React.createElement(StyledLoader, __assign({}, props))));
}));
Loader.displayName = "Loader";
Loader.defaultProps = {
    size: 'S',
    visible: true
};
export default Loader;
