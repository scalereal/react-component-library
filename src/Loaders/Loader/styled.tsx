import styled from 'styled-components';
import { Props } from '../types';

export const StyledLoader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    vertical-align: middle;
    pointer-events: none;
    border-radius: 50%;
    animation: 1s StyledLoader-01 linear infinite;

    @keyframes StyledLoader-01 {
        0% {
            transform: rotate(0deg);
        }
        100% {
            transform: rotate(360deg);
        }
    }
    
`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${StyledLoader}{
                border: 0.188em solid transparent;
            }`;
        case "XS":
            return `font-size: 1rem;
            ${StyledLoader}{
                border: 0.188em solid transparent;
            }
            `;
        case "S":
            return `font-size: 2rem;
            ${StyledLoader}{
                border: 0.2em solid transparent;
            }`;
        
        case "M":
            return `font-size: 3rem;
            ${StyledLoader}{
                border: 0.2em solid transparent;
            }`;

        case "L":
            return `font-size: 5rem;
            ${StyledLoader}{
                border: 0.2em solid transparent;
            }`;
        
        case "XL":
            return `font-size: 6rem;
            ${StyledLoader}{
                border: 0.2em solid transparent;
            }`;
  
      default:
        return `font-size: 1rem;
        ${StyledLoader}{
            border: 0.15em solid transparent;
        }`;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.display?'inline-block':'none'};
    ${StyledLoader}{
        border-left-color:  ${pr=>pr.Color?pr.Color:'inherit'};
        border-right-color:  ${pr=>pr.Color?pr.Color:'inherit'};
    }
`;
