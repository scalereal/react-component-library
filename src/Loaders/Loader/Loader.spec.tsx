import { render } from '@testing-library/react';
import * as React from 'react';
import Loader from '.';

describe('Loader component', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader
                visible={true}
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader visible={true} size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});