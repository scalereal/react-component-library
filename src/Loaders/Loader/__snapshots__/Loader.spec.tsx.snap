// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz cCCoTD"
  >
    <div
      class="sc-bdfBwQ ezQfOm"
    />
  </div>
</div>
`;

exports[`Loader component should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz lfYWDd"
  >
    <div
      class="sc-bdfBwQ ezQfOm"
    />
  </div>
</div>
`;

exports[`Loader component should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz jvGeOg"
  >
    <div
      class="sc-bdfBwQ ezQfOm"
    />
  </div>
</div>
`;
